kiosk/kiosk_browser/main_widget.py

```diff
@@ -5,17 +5,23 @@
 from PyQt5.QtWidgets import QWidget, QPushButton, QBoxLayout, QShortcut
 
 from kiosk_browser import browser_widget, captive_portal_message, connection
+from kiosk_browser import proxy as proxy_module
 
 class MainWidget(QWidget):
 
     def __init__(self, urls, toggle_sequence):
         super(MainWidget, self).__init__()
 
-        self._connection = connection.Connection(self.set_captive_portal_url)
+        proxy = proxy_module.Proxy()
+        proxy.start_monitoring_daemon()
+
+        self._connection = connection.Connection(proxy.get_current, self.set_captive_portal_url)
+        self._connection.start_monitoring_daemon()
```

Comment:
Maybe this could be named more specifically now to reflect that it is meant to detect captive portals?

Reply:
https://github.com/dividat/playos/pull/54/commits/b010e485194c6581beea971050119b690a4b6a9f